agent/config/config.go

```diff
@@ -312,6 +217,147 @@ func get(args []string, l *logrus.Entry) (cfg *Config, configFileF string, err e
 	return //nolint:nakedret
 }
 
+func setDefaults(cfg *Config, l *logrus.Entry) {
+	if cfg == nil {
+		return
+	}
+
+	// set default values
+	configureNetworkDefaults(cfg)
+
+	if cfg.WindowConnectedTime == 0 {
+		cfg.WindowConnectedTime = time.Hour
+	}
+
+	configurePaths(cfg, l)
+
+	if cfg.Server.Address != "" {
+		if _, _, e := net.SplitHostPort(cfg.Server.Address); e != nil {
+			host := cfg.Server.Address
+			cfg.Server.Address = net.JoinHostPort(host, "443")
+			l.Infof("Updating PMM Server address from %q to %q.", host, cfg.Server.Address)
+		}
+	}
+
+	// enabled cross-component PMM_DEBUG and PMM_TRACE take priority
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_DEBUG")); b {
+		cfg.Debug = true
+	}
+	if b, _ := strconv.ParseBool(os.Getenv("PMM_TRACE")); b {
+		cfg.Trace = true
+	}
+}
+
+var (
+	defaultListenSocketJSON         = "/usr/local/percona/pmm2/pmm-agent.sock"
+	defaultListenSocketGRPCFilename = "pmm-agent-grpc.sock"
+	// DefaultListenSocketGRPC defines path to a socket on which GRPC server is listening.
+	DefaultListenSocketGRPC = "/usr/local/percona/pmm2/" + defaultListenSocketGRPCFilename
+)
+
+func configureNetworkDefaults(cfg *Config) {
+	if cfg.ListenSocket == "" {
+		configureListenDefaults(cfg)
+	}
+	if cfg.ListenSocketGRPC == "" && cfg.ListenSocket != "" {
+		cfg.ListenSocketGRPC = path.Dir(cfg.ListenSocket) + "/" + defaultListenSocketGRPCFilename
+	}
```

Comment:
shouldn't these lines be part of configureListenDefaults?

Reply:
The `configureListenDefaults` are triggered only in case `ListenSocket` is empty.
I moved it to a separate function and added one more check. We had a bug there.